composer_multi_project_monitoring/main.tf

```diff
@@ -0,0 +1,1018 @@
+# [START composer_multi_project_monitoring]
+#   Monitoring for multiple Cloud Composer environments
+#   
+#   Usage:
+#       1. Create a new project that you will use for monitoring of Cloud Composer environments in other projects
+#       2. Store the name of this project in "monitoring_project" variable below, in the locals block
+#       3. Save the list of projects with Cloud Composer environments to be monitored in "monitored_projects", set in the locals block
+#       4. Run "terraform apply"
+#
+#   The script creates the following resources in the monitoring project:
+#           1. Adds monitored projects to Cloud Monitoring
+#           2. Creates Alert Policies
+#           3. Creates Monitoring Dashboard
+#
+
+
+#######################################################
+#  
+# Overall settings
+#
+########################################################
+
+
+locals {
+    monitoring_project = "YOUR_MONITORING_PROJECT"
+    monitored_projects = toset(["YOUR_PROJECT_TO_MONITOR_1", "YOUR_PROJECT_TO_MONITOR_2", "YOUR_PROJECT_TO_MONITOR_3"])
+}
+
+#######################################################
+#  
+# Provider
+#
+########################################################
+
+provider "google-beta" {
+  project = local.monitoring_project
+}
+provider "google" {
+  project = local.monitoring_project
+}
+
+#######################################################
+#  
+# Add Monitored Projects to the Monitoring project
+#
+########################################################
+
+resource "google_monitoring_monitored_project" "projects_monitored" {
+  for_each = local.monitored_projects
```

Comment:
Since multiple projects are needed, our current CI cannot e2e test these. Can we add a test config to skip this?
example: https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/cloud_run_service_add_tag/test.yaml, you can change the name to this sample name.